README.md

```diff
@@ -8,19 +8,78 @@
+* track outputs of an analysis
+* create analyses that depend on the outputs of previous analyses
+
+With `orderly` we have two main hopes:
+
+* analysts can write code that will straightforwardly run on someone else's machine (or a remote machine)
+* when an analysis that is run several times starts behaving differently it will be easy to see when the outputs started changing, and what inputs started changing at the same time
+
+`orderly` requires a few conventions around organisation of a project, and after that tries to keep out of your way.  However, these requirements are designed to make collaborative development with git easier by minimising conflicts and make backup easier by using an append-only storage system.
+
+## The problem
+
+One often-touted goal of R over point-and-click analyses packages is that if an analysis is scripted it is more reproducible.  However, essentially all analyses depend on external resources - packages, data, code, and R itself; any change in these external resources might change the results.  Preventing such changes in external resources is not always possible, but *tracking* changes should be straightforward - all we need to know is what is being used.
+
+For example, while reproducible research [has become synonymous with literate programming](https://cran.r-project.org/web/views/ReproducibleResearch.html) this approach typically increases the number of external resources.  For a typical [`knitr`](https://CRAN.R-project.org/package=knitr) document will depend on:
+
+* the source file (`.Rmd` or `.Rnw`)
+* templates used for styling
+* data that is read in for the analysis
+* code that is directly read in with `source`
+
+The `orderly` package helps by
+
+* collecting external resources before an analysis
+* ensuring that all required external resources are identified
+* remove any manual work in tracking information about these external resources
+* allow running reports multiple times and making it easy to see what changed and why
+
+The basic idea is to take the same approach a function; with a function we have a set of known inputs and a set of known outputs - but what happens inside the function is opaque.  With `orderly`, each piece of analysis will require that the user describes what is needed and what will be produced.
 
 ## The process
 
-* (optional) Establish temporary SQL views in the database
-* Run one or more queries against the database and extract a set of data
-* Run an R script against the exported tables, producing one or more report artefacts
-* Save a set of metadata
-* Archive the exported data set
+The user describes the inputs of their analysis, including:
+
+* SQL queries (if using databases)
+* Required R sources
+* External resource files (e.g., csv data files, Rmd files, templates)
+* Packages required to run the analysis
+* Dependencies on previously run analyses
+
+Along with a list of "artefacts" (file-based results) that they will produce.
```

Comment:
Might as well keep the last one in the bullet points

Reply:
except it's not an input!